lib/src/linting/rules/no-nullable-fields-within-requests.spec.ts

```diff
@@ -0,0 +1,201 @@
+  test("valid for correct usage", () => {
+    const errors = noNullableFieldsWithinRequests({
+      api: fakeLocatable<ApiNode>({
+        name: fakeLocatable("example-api")
+      }),
+      endpoints: [
+        // Endpoint with response payload
+        fakeLocatable<EndpointNode>({
+          name: fakeLocatable("listUsers"),
+          method: fakeLocatable<HttpMethod>("GET"),
+          path: fakeLocatable("/users"),
+          isDraft: false,
+          tests: [],
+          request: fakeLocatable<RequestNode>({
+            body: fakeLocatable<BodyNode>({
+              type: {
+                kind: TypeKind.STRING
+              }
+            })
+          }),
+          responses: []
+        }),
+        // Endpoint with reference type in request payload
+        fakeLocatable<EndpointNode>({
+          name: fakeLocatable("createUser"),
+          method: fakeLocatable<HttpMethod>("POST"),
+          path: fakeLocatable("/users"),
+          isDraft: false,
+          tests: [],
+          request: fakeLocatable<RequestNode>({
+            body: fakeLocatable<BodyNode>({
+              type: objectType([
+                {
+                  name: "data",
+                  type: referenceType("requestBody201", "", TypeKind.OBJECT),
+                  optional: false
+                }
+              ])
+            })
+          }),
+          responses: []
+        })
+      ],
+      types: [
+        {
+          name: "requestBody201",
+          type: objectType([
+            {
+              name: "slug",
+              type: STRING,
+              optional: true
+            }
+          ])
+        }
+      ]
+    });
+    expect(errors).toEqual([]);
+  });
+
+  test("rejects a request body object when a field is nullable instead of omittable", () => {
+    const errors = noNullableFieldsWithinRequests({
+      api: fakeLocatable<ApiNode>({
+        name: fakeLocatable("example-api")
+      }),
+      endpoints: [
+        // Endpoint with request payload
+        fakeLocatable<EndpointNode>({
+          name: fakeLocatable("listUsers"),
+          method: fakeLocatable<HttpMethod>("GET"),
+          path: fakeLocatable("/users"),
+          isDraft: false,
+          tests: [],
+          request: fakeLocatable<RequestNode>({
+            body: fakeLocatable<BodyNode>({
+              type: objectType([
+                {
+                  name: "id",
+                  type: unionType([INT32, NULL]),
+                  optional: false
+                }
+              ])
+            })
+          }),
+          responses: []
+        })
+      ],
+      types: []
+    });
+    expect(errors).toEqual([
+      {
+        message:
+          "The object type `listUsers (request body).id` defines an nullable property. Use omittable instead."
+      }
+    ]);
+  });
+
+  test("rejects a request body object when a field is nullable instead of omittable", () => {
```

Comment:
What's the difference between this test case and the previous one? (they have exactly the same description)